rkt/api_service.go

```
@@ -0,0 +1,559 @@
+		networks = append(networks, &v1alpha.Network{
+			Name: n.NetName,
+			// There will be IPv6 support soon so distinguish between v4 and v6
+			Ipv4: n.IP.String(),
+		})
+	}
+
+	return &v1alpha.Pod{
+		Id:       p.uuid.String(),
+		Pid:      int32(pid),
+		State:    state,
+		Apps:     apps,
+		Manifest: data,
+		Networks: networks,
+	}, &manifest, nil
+}
+
+func (s *v1AlphaAPIServer) ListPods(ctx context.Context, request *v1alpha.ListPodsRequest) (*v1alpha.ListPodsResponse, error) {
+	var pods []*v1alpha.Pod
+	if err := walkPods(includeMostDirs, func(p *pod) {
+		// Get basic pod info.
+		pod, manifest, err := getBasicPod(p)
+		if err != nil { // Do not return partial pods.
+			return
+		}
+
+		// Filter the pod.
+		if !filterPod(pod, manifest, request.Filter) {
+			pods = append(pods, pod)
+		}
+	}); err != nil {
+		log.Printf("Failed to list pod: %v", err)
+		return nil, err
+	}
+	return &v1alpha.ListPodsResponse{Pods: pods}, nil
+}
+
+func (s *v1AlphaAPIServer) InspectPod(ctx context.Context, request *v1alpha.InspectPodRequest) (*v1alpha.InspectPodResponse, error) {
+	uuid, err := types.NewUUID(request.Id)
+	if err != nil {
+		log.Printf("Invalid pod id %q: %v", request.Id, err)
+		return nil, err
+	}
+
+	p, err := getPod(uuid)
+	if err != nil {
+		log.Printf("Failed to get pod %q: %v", request.Id, err)
+		return nil, err
+	}
+	defer p.Close()
+
+	// Get basic pod info.
+	pod, _, err := getBasicPod(p)
+	if err != nil {
+		return nil, err
+	}
+
+	statusDir, err := p.getStatusDir()
+	if err != nil {
+		log.Printf("Failed to get pod exit status directory: %v", err)
+		return nil, err
+	}
+
+	// Fill the extra pod info that are not available in ListPods().
+	for _, app := range pod.Apps {
+		// Fill app's image info.
+		aciInfo, err := s.store.GetACIInfoWithBlobKey(app.Image.Id)
+		if err != nil {
+			log.Printf("Failed to get ACI info for image ID %q: %v", request.Id, err)
+		} else {
+			app.Image, _, _ = s.aciInfoToV1AlphaAPIImage(aciInfo)
```

We shouldn't ignore errors